PhysicsTools/TensorFlowAOT/interface/Wrapper.h

```diff
@@ -0,0 +1,141 @@
+#ifndef PHYSICSTOOLS_TENSORFLOWAOT_WRAPPER_H
+#define PHYSICSTOOLS_TENSORFLOWAOT_WRAPPER_H
+
+/*
+ * AOT wrapper interface for interacting with models compiled for different batch sizes.
+ *
+ * Author: Marcel Rieger, Bogdan Wiederspan
+ */
+
+#include <map>
+
+#include "tensorflow/compiler/tf2xla/xla_compiled_cpu_function.h"
+#include "tensorflow/core/platform/types.h"
+#include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+
+#include "PhysicsTools/TensorFlowAOT/interface/Util.h"
+
+#include "FWCore/Utilities/interface/Exception.h"
+
+namespace tfaot {
+
+  // object that wraps multiple variants of the same model, but each compiled for a different
+  // batch size, and providing access to arguments (inputs) and results (outputs) by index
+  class Wrapper {
+  public:
+    // constructor
+    explicit Wrapper(const std::string& name)
+        : name_(name), allocMode_(AllocMode::ARGS_VARIABLES_RESULTS_PROFILES_AND_TEMPS) {}
+
+    // disable copy constructor
+    Wrapper(const Wrapper&) = delete;
+
+    // disable assignment constructor
+    Wrapper& operator=(const Wrapper&) = delete;
+
+    // destructor
+    virtual ~Wrapper(){};
+
+    // getter for the name
+    inline const std::string& name() const { return name_; }
+
+    // getter for the allocation mode
+    inline AllocMode allocMode() const { return allocMode_; }
+
+    // getter for the compiled batch sizes
+    virtual const std::vector<size_t>& batchSizes() const = 0;
+
+    // returns the number of compiled batch sizes
+    inline size_t nBatchSizes() const { return batchSizes().size(); }
+
+    // returns whether a compiled model exists for a certain batch size
+    inline bool hasBatchSize(size_t batchSize) const {
+      const auto& bs = batchSizes();
+      return std::find(bs.begin(), bs.end(), batchSize) != bs.end();
```

Comment:
We would guess that the number of compiled batch sizes is generally small, but since the vector is always sorted by default, we switched to `std::binary_search` in e08b8ba.